### PR TITLE
OSDOCS-3339 - Adding network verfication content for OSD and ROSA

### PR DESCRIPTION
**Pull request by EricPonvelle**
Version(s):
`enterprise-4.12+`

Issue:
[OSDOCS-3339](https://issues.redhat.com/browse/OSDOCS-3339)

Link to docs preview:
### **OSD**

* **Introduction to OpenShift Dedicated** -> **Policies and service definition** -> **OpenShift Dedicated service definition** -> **Networking** -> [**Network verification**](http://file.rdu.redhat.com/~eponvell/05152023/openshift-dedicated/OSDOCS-3339_Network-Verification-Updates/osd_architecture/osd_policy/osd-service-definition.html#osd-network-verification_osd-service-definition)
* **Networking** -> [**Network verification**](http://file.rdu.redhat.com/~eponvell/05152023/openshift-dedicated/OSDOCS-3339_Network-Verification-Updates/networking/network-verification.html)

### **ROSA**

* **Introduction to ROSA** -> **Policies and service definition** -> **ROSA service definition** -> **Networking** -> [**Network verification**](http://file.rdu.redhat.com/~eponvell/05152023/openshift-rosa/OSDOCS-3339_Network-Verification-Updates/rosa_architecture/rosa_policy_service_definition/rosa-service-definition.html#rosa-sdpolicy-network-verification_rosa-service-definition)
* **Networking** -> [**Network verification**](http://file.rdu.redhat.com/~eponvell/05152023/openshift-rosa/OSDOCS-3339_Network-Verification-Updates/networking/network-verification.html)

QE review:
- [ ] QE has approved this change.

Additional information:
Resurrected this dead PR from #46772 that added Network Verification.

### CODE REVIEW

**Comment by kalexand-rh** [comment]
The `branch/enterprise-4.14` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.13. And any PR going into `main` must also target the latest version branch (enterprise-4.14).

If the update in your PR does NOT apply to version 4.14 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Review by jldohmann — COMMENTED** [review]
Looking really good! Minor comment and question below

File: networking/network-verification.adoc [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+a {product-title} (ROSA) 
+endif::openshift-rosa[]
+cluster into an existing Virtual Private Cloud (VPC) or create an additional machine pool with a subnet that is new to your cluster. The checks validate your network configuration and highlight errors, enabling you to resolve configuration issues prior to deployment.
+
+You can also run the network verification checks manually to validate the configuration for an existing cluster.
+
+include::modules/understanding-network-verification.adoc[leveloffset=+1]
+
+[id="scope-of-the-network-verification-checks_{context}"]
+== Scope of the network verification checks
```
[the remark this answers is not in the document]
Comment (EricPonvelle) [reply networking/network-verification.adoc:32]: I tried that then realized the reason it's in the assembly is that the last bullet for both ROSA and OSD are x-refs =\

**Pull request merged**